apps/src/lib/levelbuilder/lesson-editor/ActivitiesEditor.jsx

```diff
@@ -0,0 +1,133 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import color from '@cdo/apps/util/color';
+import ActivityCard from '@cdo/apps/lib/levelbuilder/lesson-editor/ActivityCard';
+import Activity from '@cdo/apps/templates/lessonOverview/activities/Activity';
+import {connect} from 'react-redux';
+import {addActivity} from '@cdo/apps/lib/levelbuilder/lesson-editor/activitiesEditorRedux';
+
+const styles = {
+  activityEditAndPreview: {
+    display: 'flex',
+    flexDirection: 'row'
+  },
+  editor: {
+    width: '50%'
+  },
+  preview: {
+    width: '45%',
+    marginLeft: 20
+  },
+  previewBox: {
+    border: '1px solid black',
+    padding: 10
+  },
+  addActivity: {
+    fontSize: 14,
+    color: 'white',
+    background: color.cyan,
+    border: `1px solid ${color.cyan}`,
+    boxShadow: 'none'
+  }
+};
+
+class ActivitiesEditor extends Component {
+  static propTypes = {
+    //redux
+    activities: PropTypes.array,
+    addActivity: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      targetActivitySectionPos: null
+    };
+  }
+
+  handleAddActivity = () => {
+    this.props.addActivity(
+      this.props.activities.length,
+      `activity-${this.generateActivityKey()}`
+    );
+  };
+
+  generateActivityKey = () => {
+    let activityNumber = this.props.activities.length + 1;
+    while (
+      this.props.activities.some(
+        activity => activity.key === `activity-${activityNumber}`
+      )
+    ) {
+      activityNumber++;
+    }
+
+    return activityNumber;
```

Comment:
per earlier comment:
```suggestion
    return `activity-${activityNumber}`;
```